### PR TITLE
Emc/chore/1396-get-projects-test

### PR DESCRIPTION
**Pull request by 0xemc**
## Description

- Adds test/projects/get.test.ts
- Extract schemas to `.schema`
- Extract utils to `.utils`

## Related Ticket

Resolves #1396

<!-- If there are UI changes, please include a before and after screenshot in the following template:

### CODE REVIEW

**Review by Atmosfearful — CHANGES_REQUESTED** [review]
This will conflict heavily with my changes to get so let me get that up and we can find the smoothest path to integrate

File: carbonmark-api/src/.generated/mocks/marketplace.mocks.ts [hunk 1]
```diff
@@ -307,7 +307,7 @@ export const aListing = (overrides?: Partial<Listing>, _relationshipsToOmit: Set
         leftToSell: overrides && overrides.hasOwnProperty('leftToSell') ? overrides.leftToSell! : 'excepturi cupiditate ratione quis dolorem magnam labore',
         project: overrides && overrides.hasOwnProperty('project') ? overrides.project! : relationshipsToOmit.has('Project') ? {} as Project : aProject({}, relationshipsToOmit),
         seller: overrides && overrides.hasOwnProperty('seller') ? overrides.seller! : relationshipsToOmit.has('User') ? {} as User : aUser({}, relationshipsToOmit),
-        singleUnitPrice: overrides && overrides.hasOwnProperty('singleUnitPrice') ? overrides.singleUnitPrice! : 'qui ut provident fuga perferendis qui beatae',
+        singleUnitPrice: overrides && overrides.hasOwnProperty('singleUnitPrice') ? overrides.singleUnitPrice! : 40,
```
[the remark this answers is not in the document]
Comment (0xemc) [reply carbonmark-api/src/.generated/mocks/marketplace.mocks.ts:310]: Spent some time fighting the generator to produce a string representation of a number but no bueno. I've updated the forced '

**Review by sprrwhwk — APPROVED** [review]
Great work, love the tests 👍

File: carbonmark-api/src/utils/helpers/carbonProjects.utils.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ export const fetchCarbonProject = async (args: Args) => {
  * @returns {Promise<CarbonProject[]>} An array of all fetched projects.
  */
 export const fetchAllCarbonProjects = async (): Promise<CarbonProject[]> => {
-  const [{ allProject }, { allProjectContent }] = await Promise.all([
+  const [{ allProject = [] }, { allProjectContent = [] }] = await Promise.all([
```
[the remark this answers is not in the document]
Comment (Atmosfearful) [reply carbonmark-api/src/utils/helpers/carbonProjects.utils.ts:55]: It's 'default parameter syntax'
if either of those values are `undefined`, they will instantiate an empty array instead

[the remark this answers is not in the document]
Comment (Atmosfearful) [reply carbonmark-api/src/utils/helpers/carbonProjects.utils.ts:55]: Destructure syntax needs to use the underlying property name, in this case allProject and allProjectContent are correct. If the goal is to rename the property after destructuring it, that uses a colon;
```
const { existingProp: newPropName } = obj;
```

**Pull request closed without merging**